src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -826,10 +826,8 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t, uct_iface_ops_t *tl_ops,
     status = uct_ib_mlx5_iface_select_sl(&self->super.super,
                                          &mlx5_config->super,
                                          &rc_config->super);
-    self->super.super.config.reverse_sl = (rc_config->super.reverse_sl ==
-                                           UCS_ULUNITS_AUTO) ?
-                                                  self->super.super.config.sl :
-                                                  rc_config->super.reverse_sl;
+    self->super.super.config.reverse_sl = uct_ib_iface_config_select_reverse_sl(
+            &rc_config->super, self->super.super.config.sl);
```

Comment:
can we pass just ib_config struct to this function? it has both sl and reverse_sl

Reply:
The issue is in some places this function is being called after selecting an sl:
`self->super.super.config.sl = uct_ib_iface_config_select_sl(&config->super);`

In those cases, we would want to use the selected sl and not the configured one, `self->super.super.config` is not the same struct as `rc_config->super`.

`self->super.super.config.sl` default value is 16 (an illegal SL)
while the default for `rc_config->super.sl`, coming from env configuration would be: `UCS_ULUNITS_AUTO`

Reply:
can we unite the selection of sl and reverse_sl to a single function that operates on ib_config and/or ib_iface?
if needed it can accept additional parameters such as mask of sl with AR, on transports that support it